Userland/Libraries/LibWeb/Painting/PaintableBox.cpp

```diff
@@ -280,25 +280,28 @@ Optional<PaintableBox::ScrollbarData> PaintableBox::compute_scrollbar_data(Scrol
         return {};
     }
 
+    bool is_horizontal = direction == ScrollDirection::Horizontal;
+
     auto padding_rect = absolute_padding_box_rect();
     auto scrollable_overflow_rect = this->scrollable_overflow_rect().value();
-    auto scroll_overflow_size = direction == ScrollDirection::Horizontal ? scrollable_overflow_rect.width() : scrollable_overflow_rect.height();
-    auto scrollport_size = direction == ScrollDirection::Horizontal ? padding_rect.width() : padding_rect.height();
+    auto scroll_overflow_size = is_horizontal ? scrollable_overflow_rect.width() : scrollable_overflow_rect.height();
+    auto scrollport_size = is_horizontal ? padding_rect.width() : padding_rect.height();
     if (scroll_overflow_size == 0)
         return {};
 
-    auto min_thumb_length = min(scrollport_size, 24);
-    auto thumb_length = max(scrollport_size * (scrollport_size / scroll_overflow_size), min_thumb_length);
+    auto scrollbar_rect_length = is_horizontal ? scrollport_size - scrollbar_thumb_thickness : scrollport_size;
```

Comment:
why scrollbar_thumb_thickness is subtracted only for horizontal thumb? wouldn't we need the same for vertical?

Reply:
Yes, only if you leave an empty square at the bottom-right corner where both scrollbars meet. I'm not doing that. Vertical scrollbar has no changes in this patch. Only the horizontal scrollbar is getting changes here.